latch_cli/utils/workflow.py

```diff
@@ -74,11 +34,11 @@ def _override_task_status(status: str) -> None:
             }
         """),
         {
-            "argToken": token,
-            "argNodeName": node_name,
-            "argRetry": retry,
-            "argArrIndex": arr_index,
-            "argArrRetry": arr_retry,
+            "argToken": task_id.token,
```

Comment:
I don't see you wait for the crd to get update successfully anywhere. How are you making sure that the task doesn't reschedule and start before then

Reply:
We aren't updating the CRD, instead we read from vacuole in flytepropeller. I tried going the CRD update route but because we cache a portion of the CRD in memory, the updates don't get picked up by propeller. We could invalidate the cache when there are overrides, but I found that to be much more complicated than just reading from vacuole. 

This way we eliminate the race between the resource update and the task execution because the pre-execution task always runs before the actual execution and the task won't complete until the resource overrides field in vacuole has been updated.